src/zarr/core/indexing.py

```diff
@@ -34,7 +34,7 @@
     from zarr.core.common import ChunkCoords
 
 IntSequence = list[int] | npt.NDArray[np.intp]
-ArrayOfIntOrBool = npt.NDArray[np.intp] | npt.NDArray[np.bool_]
+ArrayOfIntOrBool = IntSequence | npt.NDArray[np.bool_]
```

Comment:
This precedes this PR and is out of scope here, but I don't think we need types like `AorB` when we can just do `A | B`